### DISCUSSION TITLE
Aptos Testnet Indexer API

### DISCUSSION THREAD

**Discussion by vvtri** [aptos-labs/aptos-developer-discussions#301] [Questions] [answered]
### Discord user ID

vvtri

### Describe your question in detail.

I have checked the indexer graphql API but can't find any way to get transaction hash. Can I get transaction hash from indexer, or I can only get transaction's version from it?

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by kshitijc1506** [comment, marked as answer]
Hi @vvtri, as of now, the indexer doesn't include the transaction hash directly. I'm not sure exactly why right now, and we will get back to you!

Here's some alternative solutions for now: 
- For a known signer, we can calculate the transaction hash off-chain with the SDK. Here's an [example](https://github.com/aptos-labs/aptos-ts-sdk/blob/5cec8cecb6684ec37d379a62ae8df6a6fab06f61/tests/e2e/transaction/transactionBuilder.test.ts#L559-L579).
- For a random signer, we can get the transaction version (from indexer) and use it to query [fullnode](https://fullnode.devnet.aptoslabs.com/v1/spec#/operations/get_transaction_by_version) to output the transaction hash.

**Reply by kshitijc1506** [reply]
I'll send an update here once the indexer team add the `transaction_hash` to `user_transactions`.
